src/components/TaxPayer/SpouseAndDependent.test.tsx

```diff
@@ -0,0 +1,250 @@
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+
+    const addButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    fireEvent.click(addButton)
+
+    // forms and labels appear after clicking the add button
+    screen.getByText('First Name and Initial')
+    screen.getByText('Last Name')
+    screen.getByText('SSN / TIN')
+
+    screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+  })
+  it('saves and edits a spouse', async () => {
+    render(
+    <Provider store={store}>
+      <SpouseInfo />
+    </Provider>
+    )
+
+    const addButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    fireEvent.click(addButton)
+
+    const createButton = screen.getByRole('button', {
+      name: /Add/
+    })
+
+    screen.getByRole('button', {
+      name: /Close/
+    })
+
+    // get all three text inputs
+    const inputs = screen.getAllByRole('textbox')
+    const [firstNameInput, lastNameInput, ssidInput] = inputs
+
+    // add values for each input
+    userEvent.type(firstNameInput, 'Sally K')
+    userEvent.type(lastNameInput, 'Ride')
+    fireEvent.change(ssidInput, { target: { value: '123456789' } })
+
+    // click the create button
+    fireEvent.click(createButton)
+
+    // expect first and last name to be concatenated
+    await screen.findByText('Sally K Ride')
+    // expect ssid to appear without hyphens
+    screen.getByText('123456789')
+
+    const editButton = screen.getByLabelText('edit')
+
+    // click the edit button
+    fireEvent.click(editButton)
+
+    // get all the inputs again, which should be filled this time
+    const filledInputs = await screen.findAllByRole('textbox')
+    const [filledFirstName, filledLastName, filledSsid] = filledInputs as HTMLInputElement[]
+
+    // expect the edit button to no longer be in the document
+    expect(editButton).not.toBeInTheDocument()
+
+    // assert that the input values match what was entered
+    expect(filledFirstName.value).toBe('Sally K')
+    expect(filledLastName.value).toBe('Ride')
+    expect(filledSsid.value).toBe('123-45-6789')
+
+    // delete the old values and add new ones
+    userEvent.type(filledFirstName, '{selectall}{del}Fella')
+    userEvent.type(filledLastName, '{selectall}{del}McGee')
+    fireEvent.change(filledSsid, { target: { value: '987654321' } })
+
+    // wait for redux-persist to do some async stuff
+    await waitFor(() => {})
+
+    // click the add button to save the new values
+    fireEvent.click(screen.getByRole('button', {
+      name: /Add/
+    }))
+
+    // expect the new names to be concatenated and new ssid to appear without hyphens
+    await screen.findByText('Fella McGee')
+    screen.getByText('987654321')
+
+    const deleteButton = screen.getByLabelText('delete')
+
+    // click the delete button
+    fireEvent.click(deleteButton)
```

Comment:
This bit down here is where I realised that mocking localStorage in `setupTests` is not enough; `redux-persist` needs to be mocked as well, as the data stored in localStorage is not being wiped and state is persisting between tests as a result

Reply:
Is there a fixture that could be used? I looked around for a minute and it appears there are fixture libraries that would let you reset local storage before / after each test, that might work?

Reply:
Maybe. I think it's more about mocking the redux store and being able to clear the persisted cache. I tried a mocked store, which initially worked fine, but wouldn't update beyond the default state.

IMO it's probably worth parking for now until a proper testing strategy is defined; maybe we don't want to write 2:1 tests:code like I did here 😅